src/plugins/navigation/next.js

```diff
@@ -163,25 +163,48 @@ export default pluginFactory({
                     unansweredOnly: unansweredOnly
                 });
 
-                if (warningHelper.shouldWarnBeforeEnd()) {
+                if (warningHelper.shouldWarnBeforeEndPart()) {
+                    testRunner.trigger(
+                        'confirm.endTestPart',
+                        messages.getExitMessage(
+                            warningScope,
+                            testRunner
+                        ),
+                        triggerNextAction, // if the test taker accept
+                        enableNav, // if he refuse
+                        {
+                            buttons: {
+                                labels: {
+                                    ok: __('SUBMIT THIS PART'),
+                                    cancel: __('CANCEL')
+                                }
+                            }
+                        }
+                    );
+                } else if (warningHelper.shouldWarnBeforeEnd()) {
```

Comment:
Here should be `You will not be able to return to this test after you submit your answers. ` in `messages.getExitMessage`